lib/network_interface/worker.ex

```diff
@@ -13,45 +13,105 @@
 # limitations under the License.
 
 defmodule Nerves.NetworkInterface.Worker do
+  @moduledoc """
+  Working for NetworkInterface. 
+  See `Nerves.NetworkInterface` for more details.
+  """
+
   use GenServer
   require Logger
 
-  @moduledoc false
+  defstruct port: nil
+
+  @typedoc "State of the GenServer"
+  @type state :: %__MODULE__{port: port}
 
-  defstruct port: nil,
-            requests: []
+  @typedoc "Setup options."
+  @type options :: map 
 
+  @typedoc "Interface name"
+  @type ifname :: binary
+  
+  @spec start_link() :: GenServer.on_start()
   def start_link() do
     GenServer.start_link(__MODULE__, [], name: __MODULE__)
   end
 
+  @spec stop() :: :ok
   def stop() do
     GenServer.cast(__MODULE__, :stop)
   end
 
+  @spec interfaces() :: [ifname] 
   def interfaces() do
     GenServer.call(__MODULE__, :interfaces)
   end
 
+  @typep mac_address :: bitstring 
+
+  @type stats :: %{
+    collisions: number, 
+    multicast: number, 
+    rx_bytes: number,
+    rx_dropped: number, 
+    rx_errors: number, 
+    rx_packets: number, 
+    tx_bytes: number,
+    tx_dropped: number, 
+    tx_errors: number, 
+    tx_packets: number
+  }
+  
+  @typedoc "Status response"
+  @type status :: %{
+    ifname: ifname,
+    type: :ethernet | atom, # not sure of other types here?
```

Comment:
I've always seen that in `ifconfig` and friends, but have never seen if it changes to anything. Will update.